infra/blueprint-test/pkg/binary/bin.go

```diff
@@ -0,0 +1,58 @@
+package binary
+
+import (
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/utils"
+	"github.com/gruntwork-io/terratest/modules/logger"
+	"github.com/mitchellh/go-testing-interface"
+)
+
+type BinaryCfg struct {
+	binary string         // path to  binary
+	logger *logger.Logger // custom logger
+	dir    string         // dir to execute commands in
+}
+
+type BinaryOption func(*BinaryCfg)
+
+func WithBinary(binary string) BinaryOption {
+	return func(f *BinaryCfg) {
+		f.binary = binary
+	}
+}
+
+func WithLogger(logger *logger.Logger) BinaryOption {
+	return func(f *BinaryCfg) {
+		f.logger = logger
+	}
+}
+
+func WithDir(dir string) BinaryOption {
+	return func(f *BinaryCfg) {
+		f.dir = dir
+	}
+}
+
+// NewBinaryConfig sets defaults and validates configuration for running a binary
+func NewBinaryConfig(t testing.TB, bin string, opts ...BinaryOption) *BinaryCfg {
+	b := &BinaryCfg{binary: bin, logger: utils.GetLoggerFromT()}
+	for _, opt := range opts {
+		opt(b)
+	}
+	err := utils.BinaryInPath(b.binary)
+	if err != nil {
+		t.Fatalf("unable to find binary %s in path: %v", b.binary, err)
+	}
+	return b
+}
+
+func (b *BinaryCfg) GetBinary() string {
```

Comment:
We probably shouldn't use getters/setters. If these properties should be public, simply export them.